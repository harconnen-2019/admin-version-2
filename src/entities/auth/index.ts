export * from './api';
export { AuthProvider } from './context/auth-provider';
export { useAuth } from './hooks/use-auth';
