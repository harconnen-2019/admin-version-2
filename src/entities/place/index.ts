export { placeApi } from './api/place-api';
export * as placeType from './api/types';
export * from './hooks/use-places-query';
export { FormPlaceAdd } from './ui/form/form-place-add';
export { SelectPlaceButton } from './ui/select/select-place-button';
export { TrPlace } from './ui/tr/tr-place';
