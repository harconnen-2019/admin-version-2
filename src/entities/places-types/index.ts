export * as placeTypesType from './api/types';
export { SelectPlaceTypes } from './ui/select/select-place-types';
