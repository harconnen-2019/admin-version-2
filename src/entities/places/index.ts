export * from './api/initial';
export * as placeType from './api/types';
export * from './hooks/use-places-query';
export { FormPlaceAdd } from './ui/form/form-place-add';
export { TablePlace } from './ui/tables/table-place';
