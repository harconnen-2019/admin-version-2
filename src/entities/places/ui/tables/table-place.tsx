import { Badge, Table } from '@mantine/core';

import { PATH_PAGE } from '@/pages/path';
import { getDate } from '@/shared/lib';

import { ButtonDell, LinkAnchor, TableData } from '@/shared/ui';
import { IPlace } from '../../api/types';
import { SelectPlaceButton } from '../select/select-place-button';

interface IProperties {
  status: string;
  error: Error | null | undefined;
  data: IPlace[];
  placeId: number | undefined;
  remove: (id: number, name: string) => void;
}

/**
 * Таблица с данными витрины
 * @param root0 пропсы
 * @param root0.data список данных
 * @param root0.error ошибка
 * @param root0.status статус загрузки
 * @param root0.placeId id витрины
 * @param root0.remove метод удаления
 * @returns JSX Element
 */
export function TablePlace({ data, error, status, placeId, remove }: Readonly<IProperties>) {
  /**
   * Формируем строки таблицы
   */
  const rows = data.map((element) => (
    <Table.Tr key={element.id}>
      <Table.Td>
        {placeId === element.id && (
          <Badge fullWidth variant="light" color="grape">
            Активная
          </Badge>
        )}
      </Table.Td>
      <Table.Td>
        <LinkAnchor to={PATH_PAGE.place.edit(element.id)}>{element.name}</LinkAnchor>
      </Table.Td>
      <Table.Td>{getDate(element.created)}</Table.Td>
      <Table.Td>{getDate(element.modified)}</Table.Td>
      <Table.Td align="right">
        <SelectPlaceButton placeId={element.id} disabled={placeId === element.id} />
      </Table.Td>
      <Table.Td align="right">
        <ButtonDell callback={() => remove(element.id, element.name)} />
      </Table.Td>
    </Table.Tr>
  ));

  return (
    <TableData
      isLoading={status === 'pending'}
      error={error}
      tableHead={[
        { id: 2, name: 'Состояние', w: 200 },
        { id: 3, name: 'Витрина', w: 'auto' },
        { id: 4, name: 'Создана', w: 170 },
        { id: 5, name: 'Изменена', w: 170 },
        { id: 6, name: ' ', w: 100 },
        { id: 7, name: ' ', w: 50 },
      ]}
      empty={(status === 'success') === (data.length === 0)}
    >
      {status === 'success' && data.length > 0 && rows}
    </TableData>
  );
}
