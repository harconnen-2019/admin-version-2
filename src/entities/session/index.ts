export { sessionContracts } from './session.contracts';
export { sessionQueries } from './session.queries';
export * as sessionTypes from './session.types';
