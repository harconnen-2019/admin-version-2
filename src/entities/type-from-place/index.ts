export { typeFromPlaceApi } from './type-from-place.api';
export { typeFromPlaceContracts } from './type-from-place.contracts';
export { typeFromPlaceQueries } from './type-from-place.queries';
export * as typeFromPlaceTypes from './type-from-place.types';
