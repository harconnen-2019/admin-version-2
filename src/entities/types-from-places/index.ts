export * as typeFromPlacesType from './api/types';
export { SelectTypeFromPlaces } from './ui/select/select-types-from-places';
