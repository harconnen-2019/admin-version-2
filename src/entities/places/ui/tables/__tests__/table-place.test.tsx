import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render, screen, within } from '@test-utils';
import { BrowserRouter } from 'react-router-dom';
import { describe, it } from 'vitest';

import { placeType } from '@/entities/places';
import { TablePlace } from '../table-place';

const data: placeType.IPlace[] = [
  {
    id: 1,
    created: new Date('1946-10-04T18:12:23.671Z'),
    modified: new Date('1948-06-03T12:04:33.301Z'),
    name: 'Clothing',
    domain: 'elated-coalition.biz',
    template: 'grounded-butter',
    favicon: 'https://avatars.githubusercontent.com/u/55877764',
    og_img: 'https://avatars.githubusercontent.com/u/81553655',
    logo_dark: 'https://avatars.githubusercontent.com/u/44465421',
    logo_light: 'https://avatars.githubusercontent.com/u/71903166',
    color_scheme: 'green',
    counter_head: 'counter_head ',
    counter_body: 'counter_body',
    thankyou_type: 'page',
    type: {
      id: 1,
      created: new Date('1959-10-08T12:09:59.495Z'),
      modified: new Date('1985-03-31T08:07:53.847Z'),
      name: 'Tools',
    },
  },
  {
    id: 2,
    created: new Date('1964-02-08T15:12:48.833Z'),
    modified: new Date('1946-06-29T01:36:25.536Z'),
    name: 'Baby',
    domain: 'perfumed-freon.org',
    template: 'extraneous-shoe-horn',
    favicon: '',
    og_img: 'https://avatars.githubusercontent.com/u/68364384',
    logo_dark: '',
    logo_light: 'https://avatars.githubusercontent.com/u/34375761',
    color_scheme: 'pink',
    counter_head: 'counter_head',
    counter_body: 'counter_body',
    thankyou_type: 'pop',
    type: {
      id: 1,
      created: new Date('1959-10-08T12:09:59.495Z'),
      modified: new Date('1985-03-31T08:07:53.847Z'),
      name: 'Tools',
    },
  },
];

const queryClient = new QueryClient();

describe('Таблица со списком витрин', () => {
  const renderComponent = () =>
    render(
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          <TablePlace
            data={data}
            error={undefined}
            placeId={2}
            remove={() => {}}
            status="success"
          />
        </BrowserRouter>
      </QueryClientProvider>,
    );

  it('Создаем "снап-шот" данных', () => {
    expect(data).toMatchSnapshot();
  });

  it('Данные появятся только если статус "success', () => {
    render(
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          <TablePlace
            data={data}
            error={undefined}
            placeId={2}
            remove={() => {}}
            status="pending"
          />
        </BrowserRouter>
      </QueryClientProvider>,
    );
    const result = screen.queryByText(/clothing/i);

    expect(result).not.toBeInTheDocument();
  });

  it('Статус активная включен у нужной витрины', () => {
    renderComponent();
    // Выбираем вторую строку
    const row2 = screen.getByRole('row', {
      name: /2 активная/i,
    });
    const active = within(row2).getByText(/активная/i);

    expect(active).toBeInTheDocument();
  });

  it('Если не приходит Id витрины из сессии статус "активная" отключен', () => {
    render(
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          <TablePlace
            data={data}
            error={undefined}
            placeId={undefined}
            remove={() => {}}
            status="success"
          />
        </BrowserRouter>
      </QueryClientProvider>,
    );
    const active = screen.queryByText(/активная/i);

    expect(active).not.toBeInTheDocument();
  });

  it('Кнопка выбора витрины активна', () => {
    renderComponent();
    // Выбираем первую строку
    const row1 = screen.getByRole('row', {
      name: /1 clothing/i,
    });
    const buttonEnabled = within(row1).getByRole('button', {
      name: /выбрать/i,
    });

    expect(buttonEnabled).not.toHaveAttribute('disabled');
  });

  it('Кнопка выбора витрины выключена', () => {
    renderComponent();
    // Выбираем вторую строку
    const row2 = screen.getByRole('row', {
      name: /2 активная/i,
    });
    const buttonDisabled = within(row2).getByRole('button', {
      name: /выбрать/i,
    });

    expect(buttonDisabled).toHaveAttribute('disabled');
  });

  it('Дата переводится в удобный формат', () => {
    renderComponent();
    const parseDate = screen.getByText(/29 июня 46 г/i);

    expect(parseDate).toBeInTheDocument();
  });

  it('Адаптивная ширина у колонки с названием', () => {
    renderComponent();
    const cellHeadName = screen.getByRole('columnheader', {
      name: /витрина/i,
    });

    expect(cellHeadName).toHaveAttribute('style', 'width: auto;');
  });
});
