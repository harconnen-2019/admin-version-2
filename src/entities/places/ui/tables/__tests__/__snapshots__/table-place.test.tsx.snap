// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Таблица со списком витрин > Создаем "снап-шот" данных 1`] = `
[
  {
    "color_scheme": "green",
    "counter_body": "counter_body",
    "counter_head": "counter_head ",
    "created": 1946-10-04T18:12:23.671Z,
    "domain": "elated-coalition.biz",
    "favicon": "https://avatars.githubusercontent.com/u/55877764",
    "id": 1,
    "logo_dark": "https://avatars.githubusercontent.com/u/44465421",
    "logo_light": "https://avatars.githubusercontent.com/u/71903166",
    "modified": 1948-06-03T12:04:33.301Z,
    "name": "Clothing",
    "og_img": "https://avatars.githubusercontent.com/u/81553655",
    "template": "grounded-butter",
    "thankyou_type": "page",
    "type": {
      "created": 1959-10-08T12:09:59.495Z,
      "id": 1,
      "modified": 1985-03-31T08:07:53.847Z,
      "name": "Tools",
    },
  },
  {
    "color_scheme": "pink",
    "counter_body": "counter_body",
    "counter_head": "counter_head",
    "created": 1964-02-08T15:12:48.833Z,
    "domain": "perfumed-freon.org",
    "favicon": "",
    "id": 2,
    "logo_dark": "",
    "logo_light": "https://avatars.githubusercontent.com/u/34375761",
    "modified": 1946-06-29T01:36:25.536Z,
    "name": "Baby",
    "og_img": "https://avatars.githubusercontent.com/u/68364384",
    "template": "extraneous-shoe-horn",
    "thankyou_type": "pop",
    "type": {
      "created": 1959-10-08T12:09:59.495Z,
      "id": 1,
      "modified": 1985-03-31T08:07:53.847Z,
      "name": "Tools",
    },
  },
]
`;
